### PR TITLE
Clean up tests to reduce testing pain

### PR DESCRIPTION
**Pull request by rjdeloss**

### CODE REVIEW

File: app/src/main/java/tttTDD/Java/ConsoleClient.java [hunk 1]
```diff
@@ -105,13 +106,52 @@ private int convertLocationToInput(int value) {
     private void renderBoard() {
         String[] board = game.getBoard();
         int boardSize = gameConfig.getBoardSizeConfiguration();
+        String maxInput = String.valueOf(boardSize * boardSize);
+
+        String rowWithLines;
+        ArrayList<String> fullBoard = new ArrayList<>();
```
Comment (nystromb) [inline app/src/main/java/tttTDD/Java/ConsoleClient.java:112]: I think this code is a bit hard to follow and could be cleaner. But I'm not sure how it actually looks.

An ArrayList feels like an inefficient data structure here. When building strings, it's recommended to use the StringBuilder class, which might help keep things a bit cleaner since it provides a nice interface for some of the things you're doing here (instead of calling `join` on the string array)

https://docs.oracle.com/javase/8/docs/api/java/lang/StringBuilder.html

Comment (billwanjohi) [reply app/src/main/java/tttTDD/Java/ConsoleClient.java:112]: ooh, nice

File: app/src/main/java/tttTDD/Java/GameConfiguration.java [hunk 2]
```diff
@@ -3,12 +3,6 @@
 public class GameConfiguration {
     private int boardSize;
     private boolean isComputer;
-    private int DEFAULT_SIZE = 3;
```
Comment (joeyshipley) [inline app/src/main/java/tttTDD/Java/GameConfiguration.java:6]: Looks deleted, did you have a test that confirmed the default size was 3?

File: app/src/test/java/tttTDD/Java/BoardTest.java [hunk 3]
```diff
@@ -64,146 +64,175 @@ public void updatingTheSameSpaceWillResultInAnErrorException() {
...
-        board.updateBoard(2, "X");
-        board.updateBoard(3, "O");
-        board.updateBoard(4, "O");
-
+        updateCellsOnBoard(moves, markers);
 
         Assert.assertFalse(board.hasNoMoreMoves());
     }
+
+    private void updateCellsOnBoard(int[] locations, String[] markers) {
```
Comment (joeyshipley) [inline app/src/test/java/tttTDD/Java/BoardTest.java:216]: Nice job on these.

File: app/src/main/java/tttTDD/Java/GameConfiguration.java (hunk 2 above)
Comment (rjdeloss) [reply app/src/main/java/tttTDD/Java/GameConfiguration.java:6]: yeah I did since it was no longer being used when creating a game. Should the default constructor as well as the size still exists if it's not being called or would you say when the API interface is created, it can be brought back?

Comment (joeyshipley) [reply app/src/main/java/tttTDD/Java/GameConfiguration.java:6]: Cool, nope. You're good. I just missed it.

File: app/src/test/java/tttTDD/Java/BoardTest.java [hunk 4]
```diff
@@ -64,146 +64,175 @@ public void updatingTheSameSpaceWillResultInAnErrorException() {
...
+        updateCellsOnBoard(moves, "X");
 
         Assert.assertTrue(board.hasWinningSet("X"));
     }
 
     @Test
     public void checkIfTheBoardHasNoMoreMoves() {
         board = new Board();
+        int[] moves = {0, 1, 2, 3, 4, 5, 6, 7, 8};
+        String[] markers = {"X", "O", "X", "O", "O", "X", "O", "X", "X"};
```
Comment (joeyshipley) [inline app/src/test/java/tttTDD/Java/BoardTest.java:199]: Consideration for another day, but since the moves are linked together by array index, a parameter grouping object could serve you well here. Ie. parameters props that change or are used together are served well to live together.

Ex.
```
MoveValues[] moves = [ { Space: 0, Marker: "X" }, { Space: 1, Marker: "O" } ]
```

File: app/src/test/java/tttTDD/Java/GameTest.java [hunk 5]
```diff
@@ -35,14 +35,12 @@ public void PlayerShouldUpdateBoardWithinTheGame3x3() {
     @Test
     public void PerformTurnShouldReturnFalseWhenASpaceIsTaken() {
         game = new Game(gameConfig);
-        int humanInput1 = 0;
-        int computerInput1 = 1;
-        int humanInput2 = 0;
+        int[] humanMoves = {0, 1, 0};
+        // Moves played by computer => 1
```
Comment (joeyshipley) [inline app/src/test/java/tttTDD/Java/GameTest.java:39]: I generally don't like comments, but you did a good trade-off here. You needed to improve the usability of tests and lost some information, the important information came back as a comment. I would have done the same, then in the future, if you see an opportunity to pull the info back out of comments and get it into the code again it's always worth the refactor effort. :)

**Review by joeyshipley — APPROVED** [review]

File: app/src/test/java/tttTDD/Java/BoardTest.java (hunk 4 above)
Comment (nystromb) [reply app/src/test/java/tttTDD/Java/BoardTest.java:199]: Yeah, extracting out would add a bunch of code. You could have an array of arrays to start, but I think its a good improvement from what it was

There's probably a data structure that you can use without having to handroll your own java objects. Could be `Tuple<int, String>`, or a Pair<int, String>`, but you might have to import a library to get these

**Review by nystromb — APPROVED** [review]

File: app/src/test/java/tttTDD/Java/BoardTest.java (hunk 4 above)
Comment (joeyshipley) [reply app/src/test/java/tttTDD/Java/BoardTest.java:199]: Ah, that makes sense. Cool to hear you felt the same way. :)
